src/components/header-system/README.md

```diff
@@ -1,14 +1,384 @@
+                </div>
+
+                <div class="rvt-header-global__controls" data-rvt-disclosure="menu">
+                    
+                  <nav aria-label="Primary navigation" class="rvt-header-menu" data-rvt-disclosure-target="menu" hidden>
+                    <!-- Primary navigation... -->
+                  </nav>
+
+                  <!-- Search -->
+
+                  <div data-rvt-disclosure="search">
+                      
+                      <!-- Button that shows/hides the search field -->
+                      
+                      <button class="rvt-global-toggle" data-rvt-disclosure-toggle="search" aria-expanded="false">
+                          <span class="rvt-sr-only">Search</span>
+                          <svg xmlns="http://www.w3.org/2000/svg" class="rvt-global-toggle__search" height="16" viewBox="0 0 16 16" width="16">
+                              <path d="M15.71,14.29,10.89,9.47a6,6,0,1,0-1.42,1.42l4.82,4.82a1,1,0,0,0,1.42,0A1,1,0,0,0,15.71,14.29ZM6,10a4,4,0,1,1,4-4A4,4,0,0,1,6,10Z" fill="currentColor"></path>
+                          </svg>
+                      </button>
+
+                      <!-- Search field -->
+
+                      <form action="/search" class="rvt-header-global__search" data-rvt-disclosure-target="search" role="search" method="get" hidden>
+                          <label class="rvt-sr-only" for="search">Search</label>
+                          <div class="rvt-input-group">
+                              <input class="rvt-input-group__input rvt-text-input" type="text" name="q">
+                              <div class="rvt-input-group__append">
+                                  <button class="rvt-button">Search</button>
+                              </div>
+                          </div>
+                      </form>
+                  </div>
+                </div>
+            </div>
+        </div>
+    </div>
+</header>
+```
+
+### Implementing search
+
+By default, the user is sent to the URL `/search` when they submit a search query. The value of their search query is appended to the page URL as the query string parameter `q`.
+
+```
+GET /search?q={search query}
+```
+
+It is up to you to implement a method for handling search queries on your website or application &mdash; most sites use [Google Custom Search](https://developers.iu.edu/resources/google-custom-search/).
+
+If you'd like search to function differently, you can:
+
+- Change the value of the `action` attribute on the search `form` element to redirect the user to a different page on submit 
+- Change the value of the `method` attribute on the search `form` element to `post` to submit the search query via `HTTP POST` instead
+- Change the value of the `name` attribute on the search `input` element to use a different name for the variable containing the search query
+
+To handle submitted search queries yourself without redirecting the user to a new page, use the following JavaScript:
```

Comment:
More of just a comment here, but I'm wondering if we need to really go into detail about search implementation specific to JavaScript?

I'm thinking specifically in the web apps. All we need to do is give them the markup, styles and the show/hide logic, and then the search implementation will very much depend on their situations. Some teams might just straight handle it all on the server. In front-end framework world I would imagine they'll handle it the way that fits with their framework.

In the iu web framework/websites world I would imagine the go to implementation will remain Google custom search.

I think it's fine to leave for now. Just wanted to get your thoughts.

Reply:
@levimcg Looking back at the "Implementing search" section with fresh eyes, I'm thinking of just getting rid of it. Presumably anyone building a site or app with Rivet understands basic HTML, can look at the markup for search, and understand how it's working. I can't imagine a developer using Rivet who isn't familiar with form HTML.